Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceShaderResourceGroup.h

```diff
@@ -0,0 +1,87 @@
+    class MultiDeviceShaderResourceGroup : public MultiDeviceResource
+    {
+        friend class MultiDeviceShaderResourceGroupPool;
+
+    public:
+        AZ_CLASS_ALLOCATOR(MultiDeviceShaderResourceGroup, AZ::SystemAllocator, 0);
+        AZ_RTTI(MultiDeviceShaderResourceGroup, "{6C1B42AA-51A9-482F-9203-6415CA9373B7}", MultiDeviceResource);
+        AZ_RHI_MULTI_DEVICE_OBJECT_GETTER(ShaderResourceGroup);
+        MultiDeviceShaderResourceGroup() = default;
+        virtual ~MultiDeviceShaderResourceGroup() override = default;
+
+        using CompileMode = ShaderResourceGroup::CompileMode;
+
+        //! Compiles the SRG with the provided data.
+        //! When using Async compile mode, it queues a request that the parent pool compile this group (compilation is deferred).
+        //! When using Sync compile mode the SRG compilation will happen immediately.
+        void Compile(const MultiDeviceShaderResourceGroupData& shaderResourceGroupData, CompileMode compileMode = CompileMode::Async);
+
+        //! Returns the shader resource group pool that this group is registered on.
+        MultiDeviceShaderResourceGroupPool* GetPool();
+        const MultiDeviceShaderResourceGroupPool* GetPool() const;
+
+        //! Returns the data currently bound on the shader resource group.
+        const MultiDeviceShaderResourceGroupData& GetData() const;
+
+        //! Returns the binding slot specified by the layout associated to this shader resource group.
+        uint32_t GetBindingSlot() const;
+
+        //! Returns whether the group is currently queued for compilation.
+        bool IsQueuedForCompile() const;
+
+        //! Resets the update mask after m_updateMaskResetLatency number of compiles
+        void DisableCompilationForAllResourceTypes();
+
+        //! Returns true if any of the resource type has been enabled for compilation.
+        bool IsAnyResourceTypeUpdated() const;
+
+        //! Returns true if a specific resource type has been enabled for compilation.
+        bool IsResourceTypeEnabledForCompilation(uint32_t resourceTypeMask) const;
+
+        //! Update the m_rhiUpdateMask for a given resource type which will ensure we will compile that type for the current frame
+        void EnableRhiResourceTypeCompilation(const MultiDeviceShaderResourceGroupData::ResourceTypeMask resourceTypeMask);
+
+        //! Reset the iteration counter to 0 for a resource type which will ensure that the given type will
+        //! be compiled for another m_updateMaskResetLatency number of Compile calls
+        void ResetResourceTypeIteration(const MultiDeviceShaderResourceGroupData::ResourceType resourceType);
+
+        //! Return the view hash stored within m_viewHash
+        HashValue64 GetViewHash(const AZ::Name& viewName);
+
+        //! Update the view hash within m_viewHash
+        void UpdateViewHash(const AZ::Name& viewName, const HashValue64 viewHash);
```

Comment:
All of these methods can be removed too. m_rhiUpdateMask was used as a way to only compile parts of an SRG that were updated. This is still happening for SD version of SRG. The MD version doesn't have to worry about this as well.

Reply:
But shouldn't it still be possible to call all of these methods for **all** SD groups?
Just as an example, `void DisableCompilationForAllResourceTypes();` just forwards the call to all SD groups tied to this MD group, the same is true for most of these methods?
Would a user not want to call this on all SD groups or what would the reasoning be to call it for some and not for others (which would entail retrieving the SD group first and then calling `DisableCompilationForAllResourceTypes();`)?